cmd/kube-controller-manager/app/certificates.go

```diff
@@ -224,3 +223,80 @@ func startRootCACertificatePublisherController(ctx context.Context, controllerCo
 	go sac.Run(ctx, 1)
 	return nil, true, nil
 }
+
+func newKubeAPIServerSignerClusterTrustBundledPublisherDescriptor() *ControllerDescriptor {
+	return &ControllerDescriptor{
+		name:     names.KubeAPIServerClusterTrustBundlePublisherController,
+		aliases:  []string{"kube-apiserver-serving-clustertrustbundle-publisher"},
+		initFunc: newKubeAPIServerSignerClusterTrustBundledPublisherController,
+	}
+}
+
+func newKubeAPIServerSignerClusterTrustBundledPublisherController(ctx context.Context, controllerContext ControllerContext, controllerName string) (controller.Interface, bool, error) {
+	rootCA, err := getKubeAPIServerCAFileContents(controllerContext)
+	if err != nil {
+		return nil, true, err
```

Comment:
Applies multiple times, we should only return `true` when we actually run the controller.

```suggestion
		return nil, false, err
```